### ISSUE TITLE
Those exploding pirates

### ISSUE THREAD

**Issue by Mervill** [tgstation/tgstation#22796]
```
ADMIN LOG: Explosion with size (0, 1, 5, 0) in area Unexplored Location (149,206,6 - JMP)
ADMIN LOG: Explosion with size (0, 1, 5, 0) in area Unexplored Location (149,205,6 - JMP)
runtime error: 
[00:46:53]Cannot execute null.handle fall().
[00:46:53]proc name: fall (/mob/living/carbon/fall)
[00:46:53]  source file: carbon.dm,229
[00:46:53]  usr: null
[00:46:53]  src: Vivian Stroble (/mob/living/carbon/human)
[00:46:53]  src.loc: null
[00:46:53]  call stack:
[00:46:53]Vivian Stroble (/mob/living/carbon/human): fall(1)
[00:46:53]Vivian Stroble (/mob/living/carbon/human): update canmove()
[00:46:53]Vivian Stroble (/mob/living/carbon/human): death(1)
[00:46:53]Vivian Stroble (/mob/living/carbon/human): death(1)
[00:46:53]Vivian Stroble (/mob/living/carbon/human): death(1)
[00:46:53]Pirate (/obj/effect/mob_spawn/human/corpse/pirate): create(null)
[00:46:53]Pirate (/obj/effect/mob_spawn/human/corpse/pirate): New(null)
[00:46:53]Pirate (/mob/living/simple_animal/hostile/pirate): death(null)
[00:46:53]Pirate (/mob/living/simple_animal/hostile/pirate): death(null)
[00:46:53]Pirate (/mob/living/simple_animal/hostile/pirate): update stat()
[00:46:53]...
[00:46:53]Pirate (/mob/living/simple_animal/hostile/pirate): adjustHealth(15, 1, 0)
[00:46:53]Pirate (/mob/living/simple_animal/hostile/pirate): adjustHealth(15, 1, 0)
[00:46:53]Pirate (/mob/living/simple_animal/hostile/pirate): handle temperature damage()
[00:46:53]Pirate (/mob/living/simple_animal/hostile/pirate): handle environment(/datum/gas_mixture/space (/datum/gas_mixture/space))
[00:46:53]Pirate (/mob/living/simple_animal/hostile/pirate): Life(2)
[00:46:53]Pirate (/mob/living/simple_animal/hostile/pirate): Life(2)
[00:46:53]Pirate (/mob/living/simple_animal/hostile/pirate): Life(2)
[00:46:53]Mobs (/datum/subsystem/mobs): fire(1)
[00:46:53]Master (/datum/controller/master): Loop()
[00:46:53]Master (/datum/controller/master): StartProcessing()
runtime error: 
[00:47:29]Cannot execute null.handle fall().
[00:47:29]proc name: fall (/mob/living/carbon/fall)
[00:47:29]  source file: carbon.dm,229
[00:47:29]  usr: null
[00:47:29]  src: Raegan Shick (/mob/living/carbon/human)
[00:47:29]  src.loc: null
[00:47:29]  call stack:
[00:47:29]Raegan Shick (/mob/living/carbon/human): fall(1)
[00:47:29]Raegan Shick (/mob/living/carbon/human): update canmove()
[00:47:29]Raegan Shick (/mob/living/carbon/human): death(1)
[00:47:29]Raegan Shick (/mob/living/carbon/human): death(1)
[00:47:29]Raegan Shick (/mob/living/carbon/human): death(1)
[00:47:29]Pirate Gunner (/obj/effect/mob_spawn/human/corpse/pirate/ranged): create(null)
[00:47:29]Pirate Gunner (/obj/effect/mob_spawn/human/corpse/pirate/ranged): New(null)
[00:47:29]Pirate Gunner (/mob/living/simple_animal/hostile/pirate/ranged): death(null)
[00:47:29]Pirate Gunner (/mob/living/simple_animal/hostile/pirate/ranged): death(null)
[00:47:29]Pirate Gunner (/mob/living/simple_animal/hostile/pirate/ranged): update stat()
[00:47:29]...
[00:47:29]Pirate Gunner (/mob/living/simple_animal/hostile/pirate/ranged): adjustHealth(15, 1, 0)
[00:47:29]Pirate Gunner (/mob/living/simple_animal/hostile/pirate/ranged): adjustHealth(15, 1, 0)
[00:47:29]Pirate Gunner (/mob/living/simple_animal/hostile/pirate/ranged): handle temperature damage()
[00:47:29]Pirate Gunner (/mob/living/simple_animal/hostile/pirate/ranged): handle environment(/datum/gas_mixture (/datum/gas_mixture))
[00:47:29]Pirate Gunner (/mob/living/simple_animal/hostile/pirate/ranged): Life(2)
[00:47:29]Pirate Gunner (/mob/living/simple_animal/hostile/pirate/ranged): Life(2)
[00:47:29]Pirate Gunner (/mob/living/simple_animal/hostile/pirate/ranged): Life(2)
[00:47:29]Mobs (/datum/subsystem/mobs): fire(1)
[00:47:29]Master (/datum/controller/master): Loop()
[00:47:29]Master (/datum/controller/master): StartProcessing()
```

While we're here, does anyone know why these pirates are so prone to exploding?

**Comment by Jordie0608** [comment]
Because they have fragile hearts and explosive personalities.

**Comment by lzimann** [comment]
The problem is that some of these pirates are spawning in null space. I don't think the explosion has anything to do with it.

**Comment by Supermichael777** [comment]
it looks like they were falling into a pit while on fire?

**Comment by Shadowlight213** [comment]
handle_fall() doesn't have anything to do with pits. It has to do with lying down.
I'm going to guess that the issue is due to the pirate loot corpse being spawned after the pirate is sent to nullspace during qdel. What's odd, is I could have sworn we fixed this issue before.

**Comment by timothymtorres** [comment]
Apparently the fix for this at #23892 was being worked on but was abandoned.

**Comment by timothymtorres** [comment]
So I'm going to be closing this for the following reasons:

- This pirate runtime and #23863 were both reported at the same time 5 years ago.  There have been no other bug reports since then.
- There have been a lot of changes to simple mobs and pirates that have likely fixed the problem.  Some examples are #32214, #24074, #21464, #39424
- Pirates were changed to be skeletons instead of humans

If anyone sees these pirate runtimes again feel free to ping me and I will reopen both issues.
